resources/js/components/dashboard/components/IndividualSponsor.tsx

```diff
@@ -0,0 +1,351 @@
+                                result.push(rInfo);
+                            }
+                            return result;
+                        },[])
+                        this.setState({ resources: detailState, portalInfo: portalInfo, loadingDefinitions: false });
+                        return;
+                    }
+                }
+            } else {
+                toast.error("Failed to load more information");
+                this.setState({ loadingDefinitions: true });
+            }
+
+            this.setState({ loadingDefinitions: false });
+        });
+    }
+
+    private renderResourceCard(data) {
+        // Title, description, files
+        return (
+            <Layout.Section oneThird>
+            <Card title={this.capitalizeFirstLetter(data.mainType)} sectioned>
+                {
+                    (!data || data.files.length === 0) ? <></> : <Image
+                    // source = {logoUrl}
+                    source={data.files[0].url}
+                    alt={data.files[0].name}
+                    style={{maxWidth:"100%",maxHeight:"100%"}}
+                />
+                }
+
+                <DisplayText>{data.title ? this.capitalizeFirstLetter(data.title) : "A resource!" }</DisplayText>
+                <p>{data.description ? this.capitalizeFirstLetter(data.description) : "" }</p>
+            </Card>
+            </Layout.Section>
+        );
+    }
+
+    private invalidSponsor = () => {
+        return <Dashboard404 />;
+    };
+
+    private retrieveSponsor = (SponsorId: number) => {
+        this.setState({ loading: true });
+        axios.get(`/sponsors/dashboard-api/get-sponsors-reduced.json`).then(res => {
+            const status = res.status;
+            if(status >= 200 && status <= 300) {
+                const payload = res.data;
+                if("success" in payload && payload["success"]) {
+                    var total = payload["data"].length;
+                    const target: ISponsor = payload["data"].find(sponsor =>{
+                        return sponsor.id === SponsorId
+                    })
+                    this.setState({ 
+                        loading: false, 
+                        SponsorId: SponsorId,
+                        Sponsor: target,
+                        totalNo: total,
+                    });
+                    //{"id":2,"name":"Chuen","slug":"chuen","tier":"Chuen2","privileges":";swag;resources;workshop;social_media;mentors[42];recruiters[42]","created_at":"2020-10-31 15:27:29","updated_at":"2020-10-31 15:28:08"}
+                    return;
+                } else {
+                    console.log(payload);
+                    toast.error(payload["message"]);
+                }
+            } else {
+                toast.error("Failed to load sponsors");
+                this.setState({ loading: false });
+            }
+        })
+    }
+}
+
+export default withRouter(IndividualSponsor);
```

Comment:
I don't think you need `withRouter()` here, just `IndividualSponsor` seems to work, but I might be wrong!

Reply:
We do need it!